shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.jsonApiException
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
+    val isSuccess: Boolean = false,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    @Suppress("EmptySecondaryConstructor")
+    constructor() : this(false, false, null) {}
+
+    @Suppress("EmptySecondaryConstructor")
+    constructor(error: String?) : this(false, false, error) {}
```

Comment:
Can I know the reason why we need these constructors?

Reply:
> These empty constructor will be used on the Swift side only, because on Swift we can't call Constructor with default values. 

I decided not to add comment for Swift specific Construtors.

Please see in [this](https://github.com/blyscuit/kmm-swiftUI-IC/pull/53#discussion_r983069469) conversation 🙏